### DISCUSSION TITLE
Extension icon shows 'ERR' badge

### DISCUSSION THREAD

**Discussion by Tback1** [coddingtonbear/obsidian-web#157] [Bug Report]
### Describe the bug

always show error on icon,
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/d836db33-9c01-4930-84bc-45cb99539b19)
but connect to REST well
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/65af1e1c-312f-47bf-aa20-78ba6b4e18e3)

**Can save any page, but don't show the page has in the Vault in the pop, even it had stored in the Vault before.**

### Operating System

Windows 10 LTSC 2021

### Browser

Chrome 

### Browser Version

120.0.6099.217 X64 
Obisidian 1.5.3
obsidian-web 3.1
Local REST API 2.0.4

**Comment by coddingtonbear** [comment]
First -- please create a separate discussion for your configuration upload problem; it's never a good idea to mix two distinct problems in a single discussion or issue because it makes it a lot harder to keep the conversation on-track.  I've edited your comment to remove that second issue so that this issue has a single focus; sorry for the inconvenience!

Regarding that first problem -- the `ERR` message in the extension badge: the first step would be to collect background logs so we can see what the ERRor is.  Could you go to https://github.com/coddingtonbear/obsidian-web/wiki/How-to-find-background-script-logs and follow the instructions there?

**Reply by Tback1** [reply]
I know not to mix the question,but that rule i think is for the issue, in the discuss I try to show the whole things, So I put it together, OK I'll put it in new disscuss.

``` javascript
background.js:2 TypeError: t.direct is not iterable
    at background.js:2:8061
    at Generator.next (<anonymous>)
    at i (background.js:2:6951)
（匿名） @ background.js:2
```

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/c048f7a6-a218-4689-a560-512948b53df0)

**Reply by coddingtonbear** [reply]
Ooo -- that's a very interesting result, but unfortunately there's not enough information in a screenshot or the particular message you've copy/pasted for me to do much.  If you'd have saved the log as was suggested in the instructions I gave you, I might be able to tell you more, but as it stands, I just need more information about this particular line on your log:

![image](https://github.com/coddingtonbear/obsidian-web/assets/527661/718682fe-4f6f-4cfe-86d4-ce31b7696e10)

there might be enough information if you just expand that particular entry in your log -- what I'm looking for there is the error message returned from the API (a status code of 400 means something wasn't accepted).

**Reply by Tback1** [reply]
I follow https://github.com/coddingtonbear/obsidian-web/wiki/How-to-find-background-script-logs to get the log ,bug it just give me the content, may be you need to update the manuel to guide people to get the informathin

the whole log by save as
[console-1705598399772.log](https://github.com/coddingtonbear/obsidian-web/files/13979815/console-1705598399772.log)


`at background.js:2:8061`
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/31239c4c-65bf-470a-9699-c929e76b77ec)
```js
                        for (const n of t.direct) {
                            const t = yield(0,
                            o._obsidianRequest)(a.host, a.apiKey, `/vault/${n.filename}`, {
                                method: "get",
                                headers: {
                                    Accept: "application/vnd.olrapi.note+json"
                                }
                            }, Boolean(a.insecureMode))
                              , i = null !== (r = yield t.json()) && void 0 !== r ? r : {};
                            i.frontmatter["web-badge-color"] && chrome.action.setBadgeBackgroundColor({
                                color: i.frontmatter["web-badge-color"],
                                tabId: e
                            }),
                            i.frontmatter["web-badge-message"] && (chrome.action.setBadgeText({
                                text: i.frontmatter["web-badge-message"],
                                tabId: e
                            }),
                            chrome.action.setTitle({
                                title: i.frontmatter["web-badge-message"],
                                tabId: e
                            }))
                        }
```


`at i (background.js:2:6951)`
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/c746f9af-1b65-4bf2-83a9-ee967b49b79e)
``` js
        14377: function(e, t, n) {
            "use strict";
            var r = this && this.__awaiter || function(e, t, n, r) {
                return new (n || (n = Promise))((function(o, a) {
                    function i(e) {
                        try {
                            c(r.next(e))
                        } catch (e) {
                            a(e)
                        }
```

**Reply by coddingtonbear** [reply]
Oh, you're right -- I had assumed the log would include expanded information, but you're right that it doesn't show un-expanded content.  What I need you to do is click the arrow to the left of this line and show me what content is there:

![](https://private-user-images.githubusercontent.com/527661/297821643-718682fe-4f6f-4cfe-86d4-ce31b7696e10.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2MDU5OTksIm5iZiI6MTcwNTYwNTY5OSwicGF0aCI6Ii81Mjc2NjEvMjk3ODIxNjQzLTcxODY4MmZlLTRmNmYtNGNmZS04NmQ0LWNlMzFiNzY5NmUxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExOFQxOTIxMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NWUwMzY5YzhhOTM4OTU3ZDAwYTJhM2QyMjAwMjI0MzYyZWQ4Nzk0NDNhOGRhZWFkYmNlZjlmYWVhMDVmNDE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.czJLgbaSteJLDKKT2m62AwCq231T77HPI70WZdfFmP4)

You should see that when you do that, the `...` disappears and additional content is shown in its place.

I understand that you're hoping the screenshots will help, but unfortunately they only tell me what line the error message is being reported on -- not the context of the error; not only that, but they're screenshots of the bundled & transpiled code instead of the actual code, and that's a few steps removed from the code as it exists.

**Reply by Tback1** [reply]
I do the screenshot is to tell what information I logged and where the error link jumped to. I'm sorry I'm not familiar with the js dev and web dev, so may be need more guide to know what todo.
like this case, the `...` is not exist in the error so I don't know where to recursively expand.

the net 2 pic just expand the request and response and the expand log is
[console-1705607972782.log](https://github.com/coddingtonbear/obsidian-web/files/13981139/console-1705607972782.log)

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/be7003bd-5547-426f-86b0-08548c964b05)
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/663172d4-ab0e-497f-91bd-33d3d801b87c)

pic 2 just show the error is nothing could expand

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/d7d7d931-835e-4053-a781-325fbde7afbb)

**Reply by coddingtonbear** [reply]
I'm afraid that although you've expanded lots of things up there, you've still missed the only one that I need -- I need this specific one expanded:

![](https://private-user-images.githubusercontent.com/527661/297821643-718682fe-4f6f-4cfe-86d4-ce31b7696e10.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2NzQzOTEsIm5iZiI6MTcwNTY3NDA5MSwicGF0aCI6Ii81Mjc2NjEvMjk3ODIxNjQzLTcxODY4MmZlLTRmNmYtNGNmZS04NmQ0LWNlMzFiNzY5NmUxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExOVQxNDIxMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjE5MmFlMjg3ZDY1MTUxOWI3NTNiMTQwNWRiMDE1M2JlOTJhOWU3ODJiNmJmOGFhZjYzODdhNTE5MzgxMTJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UrV5R9YbJtwCbWXvnNmi7_lLTYYWvHf_egqTiST5Vv8)

You'll be able to tell it's the right one because it has a couple visible fields including a `url` of `https://127.0.0.1:27124/search/` and a `status` of `400` --  that URL having the word `search` in it and a status of `400` is extremely important.

I know it's a subtle thing, but above you gave me a _successful_ response (see the `status: 200` in the expansion above), and for a different URL, and those aren't helpful for figuring out what's going wrong.

**Reply by owrede** [reply]
I have the same issue. I see the ERR-badge. With that nothing is working. 

I followed the instructions for getting the console log from the service worker (had to enable the developer mode BEFORE seeing that link, would be great to add that in the insttructions). My console log looks different to the one of Tback1 (I am using Chrome Version 120.0.6099.234):

```
background.js:2 TypeError: Failed to fetch
    at background.js:2:13242
    at Generator.next (<anonymous>)
    at background.js:2:11680
    at new Promise (<anonymous>)
    at background.js:2:11425
    at h (background.js:2:13001)
    at background.js:2:12688
    at Generator.next (<anonymous>)
    at background.js:2:11680
    at new Promise (<anonymous>)
(anonym) @ background.js:2
s @ background.js:2
Promise.then (async)
c @ background.js:2
i @ background.js:2
Promise.then (async)
c @ background.js:2
(anonym) @ background.js:2
(anonym) @ background.js:2
(anonym) @ background.js:
```

**Reply by Tback1** [reply]
@coddingtonbear so recursively expand this response 400 could get what you want ?

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/5a7361b3-9ace-44c6-93c0-62ff777e78a3)

**Reply by Tback1** [reply]
so this is the console jumped, hope this is what you need~
[console-1705780464699.log](https://github.com/coddingtonbear/obsidian-web/files/13999058/console-1705780464699.log)

**Comment by coddingtonbear** [comment]
Almost!  I’m looking for the `body` of this request— it should have a useful error message in there.

![IMG_1186](https://github.com/coddingtonbear/obsidian-web/assets/527661/4634eb26-7f3d-44ae-a29a-918a2126ee90)

**Reply by Tback1** [reply]
try this
[console-1705787586119.log](https://github.com/coddingtonbear/obsidian-web/files/13999299/console-1705787586119.log)

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/44898a62-d4f6-43f6-b25f-824e9787051d)

**Comment by michaelmworthington** [comment]
@coddingtonbear First off, I just installed this and it is so much more useful to have it post the note directly to Obsidian. Thank you for all the time you have put in to developing these plugins.

I was able to find the same error in the logs as others have posted. I tested in Swagger at https://coddingtonbear.github.io/obsidian-local-rest-api/#/Search/post_search_ and received this response body:

```
{
  "message": "The query you provided could not be processed.\ntag.replace is not a function",
  "errorCode": 40070
}
```

I am not too familiar with typescript, so this is a shot in the dark, but only thing I found related was https://github.com/coddingtonbear/obsidian-local-rest-api/blob/52ebcec6c7762aa59634b018ca2c079dbde8a1cc/src/requestHandler.ts#L128

I have not looked at how to debug or get logs from Obsidian, but happy to try to help provide more information if needed.

**Reply by coddingtonbear** [reply]
Ooo -- that's a very interesting lead; let me see if I can reproduce that.

**Comment by coddingtonbear** [comment]
OK!  I've pushed up a speculative fix for what _might_ be going on given the traceback @michaelmworthington posted.  Let me know if once you upgrade the package you still see this problem.  I've so far not been able to reproduce it locally.

**Reply by coddingtonbear** [reply]
A speculative fix in `obsidian-local-rest-api`, by the way -- the obsidian plugin.

**Reply by Tback1** [reply]
it‘s working，the server response had right content and has status `200` 😲

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/3324c196-c654-4957-90d0-de3821683fdf)
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/9f112bf9-4e4f-40ac-ba4d-fff17622d7aa)

**Reply by michaelmworthington** [reply]
@coddingtonbear thanks for the fix!!

I was able to get it to work.

However, I had some trouble updating the plugin from Obsidian. Long story short, I found your `toString()` change and manually applied that. I saw your GitHub Actions failed for the `2.0.5` tag. I traced the Obsidian updater back to a HTTP 404, but couldn't find the URL Obsidian was requesting. I'm guessing that is related to the GitHub Action failure.

**Reply by owrede** [reply]
I was not able to get it to work. Somehow the build failed and the deployment action for 2.0.5 did not finish succsessfully:
https://github.com/coddingtonbear/obsidian-local-rest-api/actions

**Reply by coddingtonbear** [reply]
So: that will teach me to be a little more careful about trying to squeeze in a fix when I had a few minutes before going on a vacation where I wasn't going to have internet access.  I've pushed up a new release as 2.0.6 that should install just fine.  Good luck.
